### ISSUE TITLE
Semversion faker sometimes generating invalid version

### ISSUE THREAD

**Issue by ConnorRigby** [igas/faker#125]
```
** (Version.InvalidVersionError) invalid version: "9.48.05"
    (elixir) lib/version.ex:287: Version.parse!/1
    test/test_helper.exs:11: anonymous fn/2 in :elixir_compiler_37.__FILE__/1
    (elixir) lib/enum.ex:1856: Enum.reduce_range_inc/4
    test/test_helper.exs:10: (file)
    (elixir) lib/code.ex:376: Code.require_file/2
```

**Comment by ConnorRigby** [comment]
the simplest approach would be in this function 
```
  def semver(opts \\ []) do
    allow_pre = Keyword.get(opts, :allow_pre, false)
    allow_build = Keyword.get(opts, :allow_build, false)

    pre =
      ~w(dev alpha beta rc.0 rc.1)
      |> Enum.at(Faker.random_between(0, 4))

    formats = ["0.#.#", "#.#.#", "#.##.##"]
    formats = if allow_pre, do: formats ++ ["#.#.#-#{pre}"], else: formats
    formats = if allow_build, do: formats ++ ["#.#.#+###"], else: formats
    formats = if allow_pre and allow_build, do: formats ++ ["#.#.#-#{pre}+###"], else: formats

    formats
    |> Enum.at(Faker.random_between(0, length(formats) - 1))
    |> Faker.format()
  end
```
to change  `formats = ["0.#.#", "#.#.#", "#.##.##"]` to `formats = ["0.#.#", "#.#.#"]`

I another option would be to add something like this to the chain..
```
|> fn(formatted) -> 
   case Version.parse(formatted) do 
      {:ok, %Version{}} -> formatted
      :error -> semver(opts)
   end
end.()
```

or possibly add a new function that does the above?

**Comment by igas** [comment]
@ConnorRigby @anthonator @tobyhinloopen please take a look at proposed change in https://github.com/igas/faker/pull/153/

**Comment by igas** [comment]
Closed by #153
